example/lib/data.dart

```diff
@@ -8,38 +8,62 @@ class Data {
       id: '1',
       message: "Hi!",
       createdAt: DateTime.now(),
-      sendBy: '1', // userId of who sends the message
+      sendBy: ChatUser(
```

Comment:
We shouldn't pass the object in sendBy as it will create duplicate Objects For each message. Let's Consider 500 message by 3 users only so It will create 500 instance of ChatUser for each Message. Instead we should create only 3 ChatUser Which is connected by id with each message. for Ex: Take array of ChatUser in ChatView. and pass id of user in sendBy of ChatMessage and map it internally with the ChatUserArray

Reply:
@vatsaltanna I completely understand your point. 

However using the approach you are suggesting, we will end up with unnecessarily complicated code.

The ChatUser object is so tiny, I don't see having 500 instances of ChatUser object causing any issues.
Also in the world of NO SQL database this is absolutely normal.

Can you maybe explain this from your point of view
What is the downfall of having too many ChatUser objects?

Also keep in mind that normal if you having a long scrolling list on an App, items that are not visible on the view port are recyclable.

Reply:
Well, you are not wrong. but the suggestion I have shared is not a unnecessarily complicated code. it's just few line of logical code if we implement it correctly and that's it. Np let's wait for other reviewer's opinion and then we can proceed further for this PR. BTW we highly appriciate your contributions for this functionlity in the package. Thanks for that.